pkg/types/time.go

```diff
@@ -2179,6 +2199,7 @@ func checkTimestampType(t CoreTime, tz *gotime.Location) error {
 		convertTime := NewTime(t, mysql.TypeTimestamp, DefaultFsp)
 		err := convertTime.ConvertTimeZone(tz, BoundTimezone)
 		if err != nil {
+			_, err = adjustTimestampErrForDST(tz, t.String(), mysql.TypeTimestamp, Time{t}, err)
```

Comment:
```suggestion
			_, err2 = adjustTimestampErrForDST(tz, t.String(), mysql.TypeTimestamp, Time{t})
```
And then `if err2 != nil ....`


I find this a bit clearer. Both are fine. This might make the function a bit more generic so it could be exported at some point and use from other packages if needed.